Tasks/SonarQubePostTest/PRCA/ReportProcessor.ps1

```diff
@@ -2,6 +2,80 @@ $ProjectGuidAndFilePathMap = @{}
 $ComponentKeyAndPathMap = @{}
 $ComponentKeyAndRelativePathCache = @{}
 
+
+#region Public
+
+#
+# Fetches the new issues from the json report and also adds a "relativePath" for easier consumption
+#
+# Remark: the SonarQube report uses the component guid to refer to paths
+#
+function FetchAnnotatedNewIssues
+{    
+    $sonarReportFilePath = GetSonarReportFilePath
+    if (![System.IO.File]::Exists($sonarReportFilePath))
+    {
+        throw "Could not find the SonarQube issue report at $sonarReportFilePath. Unable to post issues to the PR."
+    }
+
+    Write-Verbose "Report file found at $sonarReportFilePath"    
+        
+    CreateProjectGuidAndPathMap    
+    
+    $sonarReportFilePath = GetSonarReportFilePath
+    $json = DeserializeReport $sonarReportFilePath
+    ConstructComponentKeyAndPathMap $json
+
+    # '@' makes sure the result set is returned as an array
+    $newIssues = @($json.issues | Where {$_.isNew -eq $true})
+    Write-Host "Found $($json.issues.Count) issues out of which $($newIssues.Count) are new"
+
+    if ($($newIssues.Count) -gt 0)
+    {
+        foreach ($issue in $newIssues)
+        {
+            $filePath = GetRelativeFilePath $($issue.component)
+
+            # Add a new property in the object which stores the file path so it can be consumed directly
+            Add-Member -InputObject $issue -MemberType NoteProperty -Name "relativePath" -Value $filePath
+        }
+    }
+
+    return $newIssues
+}
+
+#endregion
+
+#region Private
+
+#
+# Returns the path to the file containing the issues that SonarQube generates when ran in issues / incremental mode
+#
+function GetSonarReportFilePath
+{ 
+    $sonarReportFolderPath = GetSonarScannerDirectory
+    $sonarReportFilePath = [System.IO.Path]::Combine($sonarReportFolderPath, "sonar-report.json")
+
+    return $sonarReportFilePath
+}
+
+function DeserializeReport
+{
+    param ([ValidateNotNullOrEmpty()][string]$sonarReportFilePath)
+    
+    Add-Type -AssemblyName "System.Web.Extensions"
+    
+    # Read sonar-report.json file as a json object
+    $sonarReportFileContent = Get-Content -Raw $sonarReportFilePath
+    $jsonSer = New-Object -TypeName System.Web.Script.Serialization.JavaScriptSerializer
+    
+    # Default value of MaxJsonLength is 2,097,152. Increasing max length by a factor of 10 to handle bigger file sizes
```

Comment:
I suppose the original author wanted to cap this at 20 MB, but I don't see any reason for this early optimization. I'll change it to max int as you suggest.